### ISSUE TITLE
Enhanced Word Document Element Manipulation

### ISSUE THREAD

**Issue by sabir-aspose** [fileformat-words/FileFormat.Words-for-.NET#3]
We're offering the following element manipulation functions:

1. **InsertBefore:** Add a new element before an existing one.
2. **InsertAfter:** Add a new element after an existing one.
3. **Append:** Attach a new element at the end.
4. **RemoveBefore:** Delete an element before an existing one.
5. **RemoveAfter:** Delete an element after an existing one.
6. **Remove:** Erase an existing element.
7. **Update:** Modify an existing element.

**Comment by sabir-aspose** [comment]
Resolved in [23.10.0](../releases/tag/v23.10.0). Public methods `InsertBefore`, `InsertAfter`, `RemoveBefore`, `RemoveAfter`, `Remove` and `Update` methods with overloads are part of `Document` class in `Openize.Words` namespace. 

Below is an example of inserting a new paragraph after every paragraph containing text "Heading" in an existing word document:

```csharp
using System;
using System.IO;
using System.Collections.Generic;
using Openize.Words;
using Openize.Words.IElements;
namespace Openize.Words.Examples
{
    class Program
    {
        static void Main(string[] args)
        {  
            Document doc = new Document("WordDocumentWithParas.docx");
            Paragraph paragraphToInsert;
            List<Paragraph> paragraphsToFind = new List<Paragraph>();
            List<Paragraph> paragraphsToInsert = new List<Paragraph>();
            int num=0;
            foreach (IElement element in doc.GetElements())
            {
               if (element is Paragraph para && para.Text.Contains("Heading"))
               {           
                   paragraphsToFind.Add(para);
                   paragraphToInsert = new Paragraph();
                   paragraphToInsert.Runs.Add(new Run { Text = $"Normal Paragraph {num + 1} inserted by FileFormat.Words" });
                   paragraphsToInsert.Add(paragraphToInsert);
                   num++;
              }
         }
         num = 0;
         foreach (Paragraph para in paragraphsToInsert)
         {
             doc.InsertAfter(para, paragraphsToFind[num]);
             num++;
         }
         doc.Save("WordDocumentWithParasInsertedAfter.docx");
      }
   }
}
```

The issue will be closed after adding docs and examples.

**Comment by openize-words** [comment]
Already resolved.
